solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -66,6 +66,12 @@ public ZkConfigSetService(SolrZkClient zkClient) {
     this.zkClient = zkClient;
   }
 
+  @Override
+  protected SolrResourceLoader createConfigSetResourceLoader(CoreDescriptor cd, String configSetName) {
```

Comment:
to understand better, why we need to override here?

Reply:
For synthetic core, the resource loader only needs to load the configset resource, it would have no access to collection specific resource - which is non existent for this new synthetic core/collection, since it does not even exists in ZK.

There are several implementation of `ConfigSetService`, and for prototype purpose, we only support this for `ZkConfigSetSrvice`